XenAdmin/Wizards/RollingUpgradeWizard/PlanActions/AutomaticBackgroundThread.cs

```diff
@@ -81,6 +81,11 @@ private void BworkerDoWork()
                                 if (planAction is UpgradeHostPlanAction)
                                 {
                                     Host hostAfterReboot = host.Connection.Resolve(new XenRef<Host>(host.opaque_ref));
+                                    if (hostAfterReboot == null)
+                                    {
+                                        log.InfoFormat("Host '{0}' cannot be resolved. Try to resolve it again with timeout", host.Name);
+                                        hostAfterReboot = (planAction as UpgradeHostPlanAction).Host;

```

Comment:
We don't need to resolve it once with XenConnection.Resolve and then again with UpgradeHostPlanAction.Host if it fails. Just use the latter.